### PR TITLE
add index_bgen docs

### PR DESCRIPTION
**Pull request by danking**
copied from 0.1

### CODE REVIEW

File: python/hail/methods/impex.py [hunk 1]
```diff
@@ -1122,6 +1122,27 @@ def import_vcf(path, force=False, force_bgz=False, header_file=None, min_partiti
 @handle_py4j
 @typecheck(path=oneof(strlike, listof(strlike)))
 def index_bgen(path):
+    """Index .bgen files. :meth:`.HailContext.import_bgen` cannot run without these
+    indices.
+
+    The index file is generated in the same directory as `path` with the
```
Comment (jbloom22) [inline python/hail/methods/impex.py]: double backticks on parameter names.

File: python/hail/methods/impex.py [hunk 2]
```diff
@@ -1122,6 +1122,27 @@ def import_vcf(path, force=False, force_bgz=False, header_file=None, min_partiti
 @handle_py4j
 @typecheck(path=oneof(strlike, listof(strlike)))
 def index_bgen(path):
+    """Index .bgen files. :meth:`.HailContext.import_bgen` cannot run without these
```
Comment (jbloom22) [inline python/hail/methods/impex.py]: Index .bgen files as required by :meth:`.HailContext.import_bgen`.

File: python/hail/methods/impex.py [hunk 3]
```diff
@@ -1122,6 +1122,27 @@ def import_vcf(path, force=False, force_bgz=False, header_file=None, min_partiti
...
+    indices.
+
+    The index file is generated in the same directory as `path` with the
+    filename of `path` appended by `.idx`.
+
+    Example
+    -------
+
+    >>> hc1.index_bgen("data/example3.bgen")
```
Comment (jbloom22) [inline python/hail/methods/impex.py]: what's with hc1?

File: python/hail/methods/impex.py [hunk 4]
```diff
@@ -1122,6 +1122,27 @@ def import_vcf(path, force=False, force_bgz=False, header_file=None, min_partiti
...
+
+    Warning
+    -------
+        While this method parallelizes over a list of BGEN files, each file is
+        indexed serially by one core. Indexing several BGEN files on a large
+        cluster is a waste of resources, so indexing should generally be done
+        as a one-time step separately from large analyses.
```
Comment (jbloom22) [inline python/hail/methods/impex.py]: ... be done once, separately from ...

File: python/hail/methods/impex.py (hunk 2 above)
Comment (tpoterba) [reply python/hail/methods/impex.py]: hail context is gone

File: python/hail/methods/impex.py [hunk 5]
```diff
@@ -1122,6 +1122,28 @@ def import_vcf(path, force=False, force_bgz=False, header_file=None, min_partiti
...
 def index_bgen(path):
+    """Index .bgen files as required by :meth:`.methods.import_bgen`.
+
+    The index file is generated in the same directory as `path` with the
+    filename of `path` appended by `.idx`.
+
+    Example
+    -------
+
+    >>> hc.index_bgen("data/example3.bgen")
```
Comment (jbloom22) [inline python/hail/methods/impex.py]: Hail context is no longer exposed. `index_bgen` is in `impex`, shouldn't this be `methods.index_bgen`?

**Pull request merged**